value.go

```diff
@@ -253,11 +250,41 @@ func newExceptionValue(s *C.mrb_state) *Exception {
 		panic("exception value init without exception")
 	}
 
+	arenaIndex := C.mrb_gc_arena_save(s)
+
 	// Convert the RObject* to an mrb_value
 	value := C.mrb_obj_value(unsafe.Pointer(s.exc))
 
+	// Retrieve and convert backtrace to []string (avoiding reflection in Decode)
+	var backtrace []string
+	mrbBacktrace := newValue(s, C.mrb_exc_backtrace(s, value)).Array()
+	for i := 0; i < mrbBacktrace.Len(); i++ {
+		ln, _ := mrbBacktrace.Get(i)
+		backtrace = append(backtrace, ln.String())
+	}
+
+	// Extract file + line from first backtrace line
+	file := "Unknown"
+	line := 0
+	if len(backtrace) > 0 {
+		fileAndLine := strings.Split(backtrace[0], ":")
+		if len(fileAndLine) >= 2 {
+			file = fileAndLine[0]
+			line, _ = strconv.Atoi(fileAndLine[1])
```

Comment:
Fair point, I'll add the capture. I think given Ruby's exception syntax it would be sufficiently impossible for this to happen but that doesn't protect against future changes to their syntax so i'll add that check.